crates/core/src/node/inner/in_memory_inner.rs

```diff
@@ -471,10 +473,22 @@ impl InMemoryNodeInner {
         } = self.run_l2_tx_raw(l2_tx.clone(), vm)?;
 
         if let ExecutionResult::Halt { reason } = result.result {
-            // Halt means that something went really bad with the transaction execution (in most cases invalid signature,
-            // but it could also be bootloader panic etc).
-            // In such case, we should not persist the VM data, and we should pretend that transaction never existed.
-            anyhow::bail!("Transaction HALT: {reason}");
+            let reason_clone = reason.clone();
+
+            let handle = tokio::runtime::Handle::current();
+            let halt_error =
+                std::thread::spawn(move || handle.block_on(reason_clone.to_halt_error()))
+                    .join()
+                    .expect("Thread panicked");
+
+            print_execution_error(&halt_error, Some(&l2_tx));
```

Comment:
I'm not 100% sure that this is the right place. IMHO, running the executor should be detached from reporting for users (e.g. I'd imagine it as whoever calls this function can process the result, and printing it for users can be one of the options).

Not sure if it plays nicely with the existing architecture, so don't propose changing it right now. But cc the current architector @itegulov just in case.

We can already see a side effect here: preparing the execution error details may require a call to some API, which is certainly not a "normal" responsibility of the executor.

Reply:
Yes, I am unsure about this as well. The intention is to be able to provide an informed error message to the user before bailing but as you described it does not fit well here. 

I am okay with removing it in this PR and circling back to it after further discussion.